File: Array/1260.Shift_2D_Grid/Shift_2D_Grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        vector<int> total;
        int tol = 0;
        for(int i = 0; i < grid.size(); ++i){
            for(int j = 0; j < grid[i].size(); ++j){
                total.push_back(grid[i][j]);
                tol++;
            }
        }
        k %= tol;
        if(k == 0) return grid;
        while(k){
            int last = total[tol - 1];
            total.pop_back();
            total.insert(total.begin(), last);
            k--;
        }
        int curr = 0;
        for(int i = 0; i < grid.size(); ++i){
            for(int j = 0; j < grid[i].size(); ++j){
                grid[i][j] = total[curr];
                curr++;
            }
        }
        return grid;
    }
};

File: Array/1260.Shift_2D_Grid/Shift_2D_Grid.py
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        total = []
        tol = 0
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                total.append(grid[i][j])
                tol += 1
        k %= tol
        for i in range(k): total.insert(0, total.pop())
        curr = 0
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                grid[i][j] = total[curr]
                curr += 1
        return grid